lib/kino/data_table.ex

```diff
@@ -43,14 +43,23 @@ defmodule Kino.DataTable do
       data. Sorting requires traversal of the whole enumerable, so it
       may not be desirable for large lazy enumerables. Defaults to `true`
 
+   * `:formatter` - a 2-arity function that is used to format the data
+     in the table. The first parameter passed is the `key` (column name) and
+     the second is the value to be formatted. When formatting column headings
+     the key is the special value `:__header__`. Defaults to the builtin
+     formatter.
```

Comment:
I would change the function to return `{:ok, string} | :default`. This way if someone wants to customize a specific value, like dates, they can do that without reimplementing the default logic for other values.

With that, the `:formatter` option wouldn't have a default value, if present we call the formatter first, otherwise we don't, in both cases we follow up with the default logic if applicable.